File: ecommerce-java/user-service/src/main/java/com/ecommerce/userservice/controller/UserController.java
package com.ecommerce.userservice.controller;

import com.ecommerce.userservice.domain.UserEntity;
import com.ecommerce.userservice.service.UserService;
import com.ecommerce.userservice.dto.RequestUserDto;
import com.ecommerce.userservice.dto.ResponseUser;
import com.ecommerce.userservice.dto.UserDto;
import lombok.RequiredArgsConstructor;
import org.springframework.core.env.Environment;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.ArrayList;
import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/")
public class UserController {

    private final Environment env;
    private final UserService userService;

    @GetMapping("/health_check")
    public String status(){
        return String.format("It's Working in User Service on Port %s", env.getProperty("local.server.port"));
    }

    @GetMapping("/welcome")
    public String welcome(){
        return env.getProperty("greeting.message");
    }

    @PostMapping("/users")
    public ResponseEntity<ResponseUser> createUser(@RequestBody RequestUserDto user){
        UserDto userDto = UserDto.requestUserToUserDto(user);
        ResponseUser responseUser = ResponseUser.userEntityToResponseUser(userService.createUser(userDto));
        return ResponseEntity.status(HttpStatus.CREATED).body(responseUser);
    }

    @GetMapping("/users")
    public ResponseEntity<List<ResponseUser>> getUsers(){
        Iterable<UserEntity> userList = userService.getUserByAll();
        List<ResponseUser> res = new ArrayList<>();

        userList.forEach(v-> res.add(ResponseUser.userEntityToResponseUser(v)));
        return ResponseEntity.status(HttpStatus.OK).body(res);
    }

    @GetMapping("/users/{userId}")
    public ResponseEntity<ResponseUser> getUser(@PathVariable String userId){
        UserEntity userEntity = userService.getUserByUserId(userId);
        return ResponseEntity.status(HttpStatus.OK).body(ResponseUser.userEntityToResponseUser(userEntity));
    }

}

File: ecommerce-kotlin/user-service/src/main/kotlin/com/example/userservice/controller/UserController.kt
package com.example.userservice.controller

import com.example.userservice.dto.RequestUserDto
import com.example.userservice.dto.ResponseUser
import com.example.userservice.dto.UserDto
import com.example.userservice.service.UserService
import org.springframework.core.env.Environment
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/")
class UserController(
    private val env: Environment,
    private val userService: UserService
) {
    @GetMapping("/health_check")
    fun status(): String = "It's Working in User service on Port ${env.getProperty("local.server.port")}"

    @GetMapping("/welcome")
    fun welcome(): String = env.getProperty("greeting.message").toString()

    @PostMapping("/users")
    fun createUser(@RequestBody requestUserDto: RequestUserDto): ResponseEntity<ResponseUser>{
        val userDto = UserDto(requestUserDto)
        val saveUser = userService.createUser(userDto)
        return ResponseEntity.status(HttpStatus.CREATED).body(ResponseUser(saveUser))
    }

    @GetMapping("/users")
    fun getUsers(): ResponseEntity<List<ResponseUser>>{
        val res = arrayListOf<ResponseUser>()
        userService.getUserByAll().forEach{
            res.add(ResponseUser(it))
        }
        return ResponseEntity.ok().body(res)
    }

    @GetMapping("/users/{userId}")
    fun getUser(@PathVariable userId: String): ResponseEntity<ResponseUser>{
        val findUser = userService.getUserByUserId(userId)
        return ResponseEntity.ok().body(ResponseUser(findUser))
    }

}